x/staking/keeper/msg_server.go

```diff
@@ -886,6 +894,46 @@ func (k msgServer) TransferTokenizeShareRecord(goCtx context.Context, msg *types
 	return &types.MsgTransferTokenizeShareRecordResponse{}, nil
 }
 
+// DisableTokenizeShares prevents an address from tokenizing any of their delegations
+func (k msgServer) DisableTokenizeShares(goCtx context.Context, msg *types.MsgDisableTokenizeShares) (*types.MsgDisableTokenizeSharesResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	delegator := sdk.MustAccAddressFromBech32(msg.DelegatorAddress)
+
+	// If tokenization is already disabled, do nothing
+	if disabled, _ := k.IsTokenizeSharesDisabled(ctx, delegator); disabled {
```

Comment:
Just like the case in `EnableTokenizeShares` below, I would recommend we throw an error here that tells the user their tokenizations are already disabled

Reply:
agreed, done